src/targets/gpu/hip_gemm_impl.cpp

```diff
@@ -0,0 +1,599 @@
+        }
+        else
+        {
+            CHECK_HIPBLAS_ERROR(hipblasLtMatrixLayoutCreate(&matB, arg_type, n, k, ldb));
+        }
+        CHECK_HIPBLAS_ERROR(hipblasLtMatrixLayoutCreate(&matC, output_type, n, m, ldc));
+
+        if(is_3inputs)
+        {
+            CHECK_HIPBLAS_ERROR(hipblasLtMatrixLayoutCreate(&matD, output_type, n, m, ldd));
+        }
+        if(num_matrices > 1)
+        {
+            CHECK_HIPBLAS_ERROR(hipblasLtMatrixLayoutSetAttribute(
+                matA, HIPBLASLT_MATRIX_LAYOUT_BATCH_COUNT, &num_matrices, sizeof(num_matrices)));
+            CHECK_HIPBLAS_ERROR(hipblasLtMatrixLayoutSetAttribute(
+                matB, HIPBLASLT_MATRIX_LAYOUT_BATCH_COUNT, &num_matrices, sizeof(num_matrices)));
+            CHECK_HIPBLAS_ERROR(hipblasLtMatrixLayoutSetAttribute(
+                matC, HIPBLASLT_MATRIX_LAYOUT_BATCH_COUNT, &num_matrices, sizeof(num_matrices)));
+
+            CHECK_HIPBLAS_ERROR(hipblasLtMatrixLayoutSetAttribute(
+                matA, HIPBLASLT_MATRIX_LAYOUT_STRIDED_BATCH_OFFSET, &a_stride, sizeof(a_stride)));
+            CHECK_HIPBLAS_ERROR(hipblasLtMatrixLayoutSetAttribute(
+                matB, HIPBLASLT_MATRIX_LAYOUT_STRIDED_BATCH_OFFSET, &b_stride, sizeof(b_stride)));
+            CHECK_HIPBLAS_ERROR(hipblasLtMatrixLayoutSetAttribute(
+                matC, HIPBLASLT_MATRIX_LAYOUT_STRIDED_BATCH_OFFSET, &c_stride, sizeof(c_stride)));
+
+            if(is_3inputs)
+            {
+                CHECK_HIPBLAS_ERROR(
+                    hipblasLtMatrixLayoutSetAttribute(matD,
+                                                      HIPBLASLT_MATRIX_LAYOUT_BATCH_COUNT,
+                                                      &num_matrices,
+                                                      sizeof(num_matrices)));
+                CHECK_HIPBLAS_ERROR(
+                    hipblasLtMatrixLayoutSetAttribute(matD,
+                                                      HIPBLASLT_MATRIX_LAYOUT_STRIDED_BATCH_OFFSET,
+                                                      &d_stride,
+                                                      sizeof(d_stride)));
+            }
+        }
+        CHECK_HIPBLAS_ERROR(hipblasLtMatmulDescCreate(
+            &hipblaslt_desc, compute_type, compute_to_hip_type(compute_type)));
+        CHECK_HIPBLAS_ERROR(hipblasLtMatmulDescSetAttribute(
+            hipblaslt_desc, HIPBLASLT_MATMUL_DESC_TRANSB, &op_A, sizeof(int32_t)));
+        CHECK_HIPBLAS_ERROR(hipblasLtMatmulDescSetAttribute(
+            hipblaslt_desc, HIPBLASLT_MATMUL_DESC_TRANSA, &op_B, sizeof(int32_t)));
+    }
+
+    ~hip_gemm_impl()
+    {
+        CHECK_HIPBLAS_ERROR(hipblasLtMatmulDescDestroy(hipblaslt_desc));
+        CHECK_HIPBLAS_ERROR(hipblasLtMatrixLayoutDestroy(matA));
+        CHECK_HIPBLAS_ERROR(hipblasLtMatrixLayoutDestroy(matB));
+        CHECK_HIPBLAS_ERROR(hipblasLtMatrixLayoutDestroy(matC));
```

Comment:
Use managed pointers.

Reply:
I was running into issues (test case failures) with using managed pointers for this. Need to look deeper into this. Any recommendations(example) on how to use managed pointers for this are welcome :).